### DISCUSSION TITLE
How to structure a location to location request with MOTIS 0.11

### DISCUSSION THREAD

**Discussion by luckyhandler** [motis-project/motis#415] [Q&A] [answered]
We currently use MOTIS 0.8 and would like to upgrade to 0.11. Our request structure looks like this right now:

```
{
      'destination': {
        'type': 'Module',
        'target': '/intermodal',
      },
      'content_type': 'IntermodalRoutingRequest',
      'content': {
        'start_type': 'IntermodalPretripStart',
        'start': {
          // location
        },
        'start_modes': [
          {
            modeType : 'FootPPR',
            mode {
              searchOptions: {
                'profile': 'default',
                'duration_limit': 900,
              }
            },
          }
        ],
        'destination_type': 'InputPosition',
        'destination': { 
          // location 
        },
        'destination_modes': [
          {
            modeType : 'FootPPR',
            mode {
              searchOptions: {
                'profile': 'default',
                'duration_limit': 900,
              }
            },
          }
        ],
        'search_type': 'Default',
        'search_dir': 'Forward',
      },
}
```

Which would be the equivalent expression for the new nigiri core? Any hints?

**Comment by felixguendling** [comment]
Probably, the `router` attribute is missing. You can set it to an empty string to use the setting `intermodal.router` from the `config.ini`.
https://github.com/motis-project/motis/blob/master/protocol/intermodal/IntermodalRoutingRequest.fbs#L131

**Reply by luckyhandler** [reply, marked as answer]
Yes, this solves the problem. Thanks. Would be great to have this as default if not set to maintain backwards compatibility...

**Reply by luckyhandler** [reply]
@felixguendling do you plan to create a patch or new version containing this fallback behaviour?

**Reply by felixguendling** [reply]
#418 will be merged soon and should make the attribute optional.

**Reply by felixguendling** [reply]
New release:
https://github.com/motis-project/motis/releases/tag/v0.11.12
